core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -100,11 +99,14 @@ public void dispatchRequest(HttpRequest request, HttpChannel channel, ThreadCont
             handleFavicon(request, channel);
             return;
         }
+        HttpChannel responseChannel = channel;
         try {
             inFlightRequestsBreaker(circuitBreakerService).addEstimateBytesAndMaybeBreak(request.content().length(), "<http_request>");
-            restController.dispatchRequest(request, new ResourceHandlingHttpChannel(channel, circuitBreakerService), threadContext);
-        } catch (CircuitBreakingException ex) {
-            restController.sendErrorResponse(request, channel, ex);
+            // iff we could reserve bytes for the request we need to send the response also over this channel
+            responseChannel = new ResourceHandlingHttpChannel(channel, circuitBreakerService);
+            restController.dispatchRequest(request, responseChannel, threadContext);
+        } catch (Throwable t) {
```

Comment:
can we clean up the other try catch? it's not needed now (as we catch things here too)